### ISSUE TITLE
Button: size inconsistent

### ISSUE THREAD

**Issue by ghost** [primefaces/primevue#6072]
### Describe the bug

Icon-only buttons do not properly change their rendered size when using the `size` prop.

As you can see, when using the Lara theme, in v4 syled mode the `rounded` icon-only buttons have the same size regardless of the value of the `size` prop.

In v3 styled mode, the `rounded` and rectangular `size="large"` icon-only buttons are a little bit wider than the other icon-only buttons.

In v3 unstyled mode, the `rounded` icon-only buttons do actually differ in size, but only the height changes. Of course `rounded` icon-only buttons should always resemble a perfect circle, not an egg.

Also in v3 unstyled mode, both `rounded` and rectangular `size="small"` icon-only buttons have a different height compared to their `label`ed counterparts. They are slightly taller than the `size="small"` ones with a `label`.

**TL;DR:**

The height of icon-only buttons should not differ compared to `label`ed buttons with the same `size`. Because they are icon-only Buttons, they should also have their width set to the same value as the height, resulting in rectangular Buttons resembling a square and `rounded` Buttons resembling a circle.

#### v4 styled

![buttons-v4-styled](https://github.com/user-attachments/assets/5a266a46-4c15-4b43-8e51-2096caf6c3b1)

#### v3 styled

![buttons-v3-styled](https://github.com/user-attachments/assets/5576c15d-a0ff-458c-a361-372f1ae488ef)

#### v3 unstyled
![buttons-v3-unstyled](https://github.com/user-attachments/assets/3cf9305a-65cd-4db3-bab3-87fc327fa326)

### Reproducer

https://stackblitz.com/~/github.com/Peanut4287/buttons-v4-styled
https://stackblitz.com/~/github.com/Peanut4287/buttons-v3-styled
https://stackblitz.com/~/github.com/Peanut4287/buttons-v3-unstyled

### PrimeVue version

3 & 4

### Vue version

3.x

### Language

TypeScript

### Build / Runtime

Vue CLI App

### Browser(s)

Chrome, Firefox

**Comment by cagataycivici** [comment]
https://github.com/primefaces/primevue/issues/7017
